### DISCUSSION TITLE
Не перехватывает youtube.com

### DISCUSSION THREAD

**Discussion by d-martynov** [xtrime-ru/antizapret-vpn-docker#16] [Q&A] [answered]
В `include-hosts-custom.txt` добавлен домен `youtube.com`, однако DNS всё равно возвращает гугловый IP.

Из контейнера:
```
nslookup youtube.com 127.0.0.1
Server:         127.0.0.1
Address:        127.0.0.1#53

Non-authoritative answer:
Name:   youtube.com
Address: 142.250.201.174
Name:   youtube.com
Address: 2a00:1450:4007:81a::200e
```

При этом другие домены из `include-hosts-custom.txt` резолвятся корректно. Например:
```
nslookup googlevideo.com 127.0.0.1
Server:         127.0.0.1
Address:        127.0.0.1#53

Non-authoritative answer:
Name:   googlevideo.com
Address: 10.224.0.14
```

В файл `knot-aliases-alt.conf` домен `youtube.com` не попадает:
```
cat /root/antizapret/result/knot-aliases-alt.conf | grep "youtube.com"
'apiyoutube.com.',
'bardforyoutube.com.',
'convyoutube.com.',
'nayoutube.com.',
'nhacmp3youtube.com.',
'taivideoyoutube.com.',
'viyoutube.com.',
'vpnforyoutube.com.',
'youtube-com-activate.ru.',
'youtube-com.club.',
```

Другие домены присутствуют. 

Вывод `doall.sh`:
```
LANG=C.UTF-8 /root/antizapret/doall.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 83.4M  100 83.4M    0     0  20.2M      0  0:00:04  0:00:04 --:--:-- 20.2M
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  589k  100  589k    0     0  1861k      0 --:--:-- --:--:-- --:--:-- 1864k
awk: cmd. line:1: warning: regexp escape sequence `\_' is not a known regexp operator
Blocked domains: 210732 result/hostlist_zones.txt
iplist_all: 152437 result/iplist_all.txt
iplist_special_range: 9 result/iplist_special_range.txt
iplist_blockedbyip: 6207 result/iplist_blockedbyip.txt
iplist_blockedbyip_noid2971: 6207 result/iplist_blockedbyip_noid2971.txt
```

Пробовал как локально собранный образ, так и из `xtrime/antizapret-vpn:latest` из реджистри.

Подскажите, в какую сторону смотреть? 
Пока меня смущает только ошибка регулярки из вывода `doall.sh`. Однако данный корректный домен ничем не отличается от других в файле.

**Comment by xtrime-ru** [comment, marked as answer]
Проблема скорее всего в файле exclude-hosts-dist.txt , в оригинальном образе там указан и youtube и другие домены гугл. 
Я исправил это в данном комите: https://github.com/xtrime-ru/antizapret-vpn-docker/commit/2223ab46ea7f51f0cf58e6531a8c7dbd9f0efe32 
Сейчас в контейенр смонтирован чистый файл, что бы убрать все исключения. 

Так же есть файл exclude-regexp-dist.awk в котором ранее была регулярка для исключения  googleusercontent. Она удалена тут: https://github.com/xtrime-ru/antizapret-vpn-docker/commit/b9b1d7804dba64b9a7c705435b7b29b8c4b7c691#diff-cd969b42790682631090a57ba517ee84806db5fbf37f940348bbd13ec2875363R6

Резюмируя: 
В последних версиях контейенра youtube гарантированно работает, никакие доработки не нужны. 
Предполагаю, что обновление было совершено как то не корректно. 

Просьба обновиться на последнюю версию **(локальные изменения будут удалены, но ключи будут работать)**
```
git fetch && git reset --hard origin/master
docker compose pull
docker compose up -d
```

После обновления нужно подождать несколько минут, пока не отработает обновление списков и проверить, что youtube резолвится в подсеть 10.224.0.0: 

```shell
docker exec antizapret-vpn dig +short youtube.com @127.0.0.4 
```

**Reply by d-martynov** [reply]
Спасибо за ответ. Монтирование всех config файлов помогло. Запускаю в кубере, не стал монтировать пустые, это и сгубило :)
